src/test/java/htsjdk/samtools/util/PositionalOutputStreamTest.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015 Daniel Gómez-Sánchez
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package htsjdk.samtools.util;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.io.OutputStream;
+
+/**
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class PositionalOutputStreamTest {
+
+    @Test
+    public void basicPositionTest() throws Exception {
+        // wrapped null output stream to check
+        final PositionalOutputStream wrapped = new PositionalOutputStream(new OutputStream() {
+            @Override
+            public void write(int b) throws IOException {}
+        });
+        // check that we start at position 0
+        Assert.assertEquals(wrapped.getPosition(), 0);
+        // check that write one int just add one
+        wrapped.write(100);
+        Assert.assertEquals(wrapped.getPosition(), 1);
+        // check that write a byte array adds its length
+        final byte[] bytes = new byte[]{1, 3, 5, 7};
+        wrapped.write(bytes);
+        Assert.assertEquals(wrapped.getPosition(), bytes.length + 1);
```

Comment:
Thanks for adding this test. One minor request for clarity - can you keep the running count in a variable so that each assert doesn't need to recalculate the running total.